cmd/torusblk/nbd.go

```diff
@@ -97,18 +97,16 @@ func connectNBD(srv *torus.Server, f *block.BlockFile, target string, closer cha
 	defer f.Close()
 	size := f.Size()
 
-	gmd, err := srv.MDS.GlobalMetadata()
-	if err != nil {
-		return err
-	}
+	gmd := srv.MDS.GlobalMetadata()
 
 	handle := nbd.Create(f, int64(size), int64(gmd.BlockSize))
 
 	if target == "" {
-		target, err = nbd.FindDevice()
+		t, err := nbd.FindDevice()
```

Comment:
Note err had to be initialized here (see the colon). `target, err := nbd.FindDevice()` would shadow the target variable from the outer scope, and its value would never be changed. It could also be resolved by declaring err at the first place, but I'm not sure whether that is clearer.